mage_integrations/mage_integrations/sources/api/__init__.py

```diff
@@ -138,51 +204,59 @@ def load_data(
         response_parser = self.config.get('response_parser')
         columns = self.config.get('columns')
 
-        response = self.__build_response()
-        result = response.json()
+        response = self._build_response()
 
-        if response_parser:
-            result = dig(result, response_parser)
+        checked_type = self._check_response_type(response)
 
-        sample = None
-        requires_columns = False
-        if result and len(result) >= 1:
-            sample = result[0]
-            requires_columns = type(sample) is not dict
+        if isinstance(checked_type, pd.DataFrame):
```

Comment:
I think you return polars dataframe in the method above. But you check whether type is pandas dataframe here.
Did you test this code path?